bottlecap

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+# bottlecap, 'cause it's kinda like cork... get it?
+
+usage() {
+	echo "usage:"
+	echo "bottlecap [--dev] [--runtime $runtime] [--build-dir $build_dir]  [--container $container] assemblerargs..."
+	echo "  --dev: rebuild and reinstall mantle, the build scripts, and the ostree-releng-scipts before running coreos-assembler"
+	echo "  --runtime: (auto) specify whether to use podman or docker. Defaults to podman if it is in \$PATH"
+	echo "  --build-dir: Where to put build artifacts, defaults to the current directory"
+	echo "  --container: which coreos-assembler container to use"
+	echo "  --help: print this help message"
+	exit 0
+}
+
+dev=0
+runtime="podman"
+which podman > /dev/null 2> /dev/null || runtime="docker"
+build_dir="$(pwd)"
+container="quay.io/cgwalters/coreos-assembler"
+
+while :
+do
+	case "$1" in
+		"--dev")
+	  		dev=1
+			shift
+			;;
+		"--runtime")
+			shift
+			runtime="$1"
+			shift
+			;;
+		"--build-dir")
+			shift
+			build_dir="$1"
+			shift
+			;;
+		"--container")
+			shift
+			container="$1"
+			shift
+			;;
+		"--help")
+			usage
+			;;
+		*) # Rest of the args are for COSA
+			break
+			;;
+	esac
+done
+
+script_dir="$(dirname $(readlink -f $0))"
+
+volumes="-v $build_dir:/srv "
+
+entrypoint=""
+
+if [ "$dev" == 1 ]; then
+	volumes+="-v $script_dir:/host/src "
+	volumes+="-v $(go env GOCACHE):/root/.cache/go-build "
```

Comment:
On a SELinux enabled system this assumes these are labeled too.  Which may be OK, but will almost certainly bite people.  (I try not to run containers accessing my `/home` for example)

Reply:
hmm. the gocache is easy enough (just put it literally anywhere that will persist across container runs).

If we want to support local dev though we need to be able to bind in wherever the build scripts are coming from (which is right now, this repo)

Reply:
The TL;DR is that SELinux + docker/podman (as root anyways) want files to be labeled specially for containers.  Trying to do `-v /home/myuser:/home` for example will fail, and you don't want to relabel your `/home`.

Reply:
(To pivot this topic though, while I haven't played with it the "rootless" podman should sidestep this issue; if we merge the bits to have this container not require privileges, just `/dev/kvm`, that will work well for rootless too)

Reply:
another note here: we need to check if `go env GOCACHE` reports a value before we try to mount it in:

```
$ ./bottlecap --build-dir /srv/ --dev --container ca build
./bottlecap: line 81: go: command not found
+ sudo podman run --rm --net=host -ti --privileged --userns=host -v /var/srv:/srv -v /var/sharedfolder/code/github.com/coreos/coreos-assembler:/host/src -v :/root/.cache/go-build --workdir /srv --entrypoint /host/src/bottlecap-shim ca d
invalid host path, must be an absolute path ""
```